pkg/controller/trigger.go

```diff
@@ -354,6 +360,56 @@ func (c *Controller) createWorkplan(wf *api.Workflow, secretRef *core.SecretEnvS
 	return wp, nil
 }
 
+func (c *Controller) resolveTemplate(wf *api.Workflow) ([]api.Step, error) {
```

Comment:
Use https://github.com/drone/envsubst/blob/master/eval.go#L6

- https://stackoverflow.com/questions/34570300/what-does-xyz-mean-in-a-shell-file